ecosystem/indexer-grpc/indexer-grpc-manager/src/data_manager.rs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{config::CacheConfig, metadata_manager::MetadataManager};
+use anyhow::{bail, ensure, Result};
+use aptos_indexer_grpc_utils::{
+    config::IndexerGrpcFileStoreConfig, file_store_operator_v2::file_store_reader::FileStoreReader,
+};
+use aptos_protos::{
+    internal::fullnode::v1::{
+        transactions_from_node_response::Response, GetTransactionsFromNodeRequest,
+    },
+    transaction::v1::Transaction,
+};
+use futures::StreamExt;
+use prost::Message;
+use std::{
+    collections::VecDeque,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+use tokio::sync::{mpsc::channel, RwLock};
+use tracing::{debug, error, info, trace, warn};
+
+struct Cache {
+    start_version: u64,
+    file_store_version: AtomicU64,
+    transactions: VecDeque<Transaction>,
+    cache_size: usize,
+
+    max_cache_size: usize,
+    target_cache_size: usize,
+}
+
+impl Cache {
+    fn new(cache_config: CacheConfig, file_store_version: u64) -> Self {
+        Self {
+            start_version: file_store_version,
+            file_store_version: AtomicU64::new(file_store_version),
+            transactions: VecDeque::new(),
+            cache_size: 0,
+            max_cache_size: cache_config.max_cache_size,
+            target_cache_size: cache_config.target_cache_size,
+        }
+    }
+
+    fn maybe_evict(&mut self) -> bool {
```

Comment:
nit: comments.

only when the file store is delayed, we'll not evict entries.

Reply:
also based on the name, it could be entried are evicted but the cache is still no usable... rename it? `cleanup` or `garbage_collect`?

Reply:
Done